src/code/z_view.c

```diff
@@ -1,62 +1,603 @@
+        view->unk_104.y += ((view->unk_E8.y - view->unk_104.y) * view->unk_100);
+        view->unk_104.z += ((view->unk_E8.z - view->unk_104.z) * view->unk_100);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA76C.s")
+        view->unk_110.x += ((view->unk_F4.x - view->unk_110.x) * view->unk_100);
+        view->unk_110.y += ((view->unk_F4.y - view->unk_110.y) * view->unk_100);
+        view->unk_110.z += ((view->unk_F4.z - view->unk_110.z) * view->unk_100);
+    }
+    Matrix_MtxToMtxF(mtx, &mtxF);
+    Matrix_Put(&mtxF);
+    Matrix_RotateX(view->unk_104.x, 1);
+    Matrix_RotateY(view->unk_104.y, 1);
+    Matrix_RotateZ(view->unk_104.z, 1);
+    Matrix_Scale(view->unk_110.x, view->unk_110.y, view->unk_110.z, 1);
+    Matrix_RotateZ(-view->unk_104.z, 1);
+    Matrix_RotateY(-view->unk_104.y, 1);
+    Matrix_RotateX(-view->unk_104.x, 1);
+    Matrix_ToMtx(mtx, "../z_view.c", 566);
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA78C.s")
+void func_800AAA50(View* view, s32 arg1) {
+    arg1 = (view->unk_120 & arg1) | arg1 >> 4;
+    if (arg1 & 8) {
+        func_800AB0A8(view);
+    } else {
+        func_800AAA9C(view);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA7AC.s")
+#ifdef NON_MATCHING
+// Just one stack var
+s32 func_800AAA9C(View* view) {
+    s32 pad;
+    s32 pad1;
+    s32 ylen;
+    Viewport* viewport;
+    Mtx* projection;
+    MtxF* viewing;
+    Gfx* gfxArr[6];
+    s32 xlen;
+    GraphicsContext* gfxCtx;
+    s32 i;
+    MtxF mtxF;
+    MtxF mtxFv;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA7B8.s")
+    gfxCtx = view->gfxCtx;
+    func_800C6AC4(gfxArr, gfxCtx, "../z_view.c", 596);
+    viewport = Graph_Alloc(gfxCtx, sizeof(Viewport));
+    LogUtils_CheckNullPointer("vp", viewport, "../z_view.c", 601);
+    func_800AA190(viewport, &view->viewport);
+    view->unk_50 = *viewport;
+    func_800AA550(view);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA814.s")
+    gSPViewport(gfxCtx->polyOpa.p++, viewport);
+    gSPViewport(gfxCtx->polyXlu.p++, viewport);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_view/func_800AA840.s")
+    projection = Graph_Alloc(gfxCtx, sizeof(Mtx));
+    LogUtils_CheckNullPointer("projection", projection, "../z_view.c", 616);
+    view->unk_E0 = projection;
+    xlen = view->viewport.rightX - view->viewport.leftX;
+    ylen = view->viewport.bottomY - view->viewport.topY;
+    if ((u16)0xB == HREG(80)) {
+        if ((u16)0xB != HREG(94)) {
+            HREG(94) = (u16)0xB;
+            HREG(83) = (u16)0x3C;
+            HREG(84) = (u16)0x3415;
+            HREG(85) = (u16)0xA;
+            HREG(86) = (u16)0x3200;
+            HREG(87) = (u16)0x64;
+        }
+        guPerspective(projection, &view->unk_11C, (f32)HREG(83), (f32)HREG(84) / 10000.0f, (f32)HREG(85), (f32)HREG(86),
+                      (f32)HREG(87) / 100.0f);
```

Comment:
Are all these (u16) / (f32) casts needed? Otherwise I would omit as many as possible.
And I think there are a few other places with possibly unnecessary casts.